packages/ember-data/lib/system/record-array-manager.js

```diff
@@ -79,7 +79,7 @@ export default Ember.Object.extend({
 
   //Don't need to update non filtered arrays on simple changes
   _recordWasChanged: function (record) {
-    var typeClass = record.constructor;
+    var typeClass = record.type;
```

Comment:
this will be a string after my PR for removing passing factories to store type gets merged.